src/l4meta/duplicates.py

```diff
@@ -0,0 +1,89 @@
+"""Find duplicates."""
+
+import os
+import stat
+import sys
+import hashlib
+
+
+def find_duplicates(directory):
+    """Find duplicates in directory."""
+    # Dups in format {hash:[names]}
+    duplicates = {}
+    for dir_name, _, filenames in os.walk(directory):
+        print('Scanning %s...' % dir_name)
+        for filename in filenames:
+            # Get the path to the file
+            path = os.path.join(dir_name, filename)
+            file_hash = calculate_hash(path)
+            # Add or append the file path
+            if file_hash in duplicates:
+                duplicates[file_hash]['path'].append(path)
+            else:
+                duplicates[file_hash] = {}
+                duplicates[file_hash]['path'] = [path]
+                if os.path.isfile(path):
+                    duplicates[file_hash]['size'] = os.stat(path)[stat.ST_SIZE]
+    return duplicates
+ 
+
+def join_dictionaries(dict1, dict2):
+    """Join two dictionaries."""
+    for key in dict2.keys():
+        dict1[key] = dict1[key] if key in dict1 else '' + dict2[key]
+
+
+def calculate_hash(path, blocksize: int = 65536):
+    """Calculate hash of a path."""
+    afile = open(path, 'rb')
+    hasher = hashlib.md5()
+    buf = afile.read(blocksize)
+    while len(buf) > 0:
+        hasher.update(buf)
+        buf = afile.read(blocksize)
+    afile.close()
+    return hasher.hexdigest()
```

Comment:
I also like that we have this subfeature to calculate hash, but a few concerns:

- `md5` is no longer secure by design. Should consider alternatives like `sha2`.
- Is it really even necessary?

Reply:
Hi Wei Shen, actually got inspiration from this website[https://www.pythoncentral.io/finding-duplicate-files-with-python/] and a few other sources. Thanks for the review, have looked online and changed `md5` to `sha256` ! As for whether it's necessary, I'm not sure if there are alternative methods of calculating hash of a given file ?

Reply:
What say you @mengwong et al? Necessary evil or overkill? 😅

Reply:
Oh but I remember a conversation with @mengwong a few months back on a separate feature and he referred me to this link: https://flylib.com/books/en/2.506.1.18/1/. It said:

> **Tenet 2: Make each program do one thing well**

So if we assume what the tenet says, then having this feature can be considered overkill. But on the other hand we should also assume that someone can somehow abuse the program and find security flaws.

I think it's really unnecessary at this stage because I don't foresee any security concerns but I'd like to get some opinions.